tests/unit/style/parsing/mask.rs

```diff
@@ -0,0 +1,121 @@
+
+    let mut parser = Parser::new("url(\"http://servo/test.png\") repeat add");
+    let result = mask::parse_value(&context, &mut parser).unwrap();
+
+    assert_eq!(result.mask_image.unwrap(), parse_longhand!(mask_image, "url(\"http://servo/test.png\")"));
+    assert_eq!(result.mask_repeat.unwrap(), parse_longhand!(mask_repeat, "repeat"));
+    assert_eq!(result.mask_composite.unwrap(), parse_longhand!(mask_composite, "add"));
+
+    let mut parser = Parser::new("intersect");
+    let result = mask::parse_value(&context, &mut parser).unwrap();
+
+    assert_eq!(result.mask_composite.unwrap(), parse_longhand!(mask_composite, "intersect"));
+
+    let mut parser = Parser::new("url(\"http://servo/test.png\")");
+    let result = mask::parse_value(&context, &mut parser).unwrap();
+
+    assert_eq!(result.mask_image.unwrap(), parse_longhand!(mask_image, "url(\"http://servo/test.png\")"));
+}
+
+#[test]
+fn mask_shorthand_should_parse_position_and_size_correctly() {
+    let url = Url::parse("http://localhost").unwrap();
+    let context = ParserContext::new(Origin::Author, &url, Box::new(CSSErrorReporterTest));
+    let mut parser = Parser::new("7px 4px");
+    let result = mask::parse_value(&context, &mut parser).unwrap();
+
+    assert_eq!(result.mask_position.unwrap(), parse_longhand!(mask_position, "7px 4px"));
+
+    let mut parser = Parser::new("7px 4px / 30px 20px");
+    let result = mask::parse_value(&context, &mut parser).unwrap();
+
+    assert_eq!(result.mask_position.unwrap(), parse_longhand!(mask_position, "7px 4px"));
+    assert_eq!(result.mask_size.unwrap(), parse_longhand!(mask_size, "30px 20px"));
+
+    let mut parser = Parser::new("/ 30px 20px");
+    assert!(mask::parse_value(&context, &mut parser).is_err());
+
+    let mut parser = Parser::new("match-source repeat-x / 30px 20px");
+    assert!(mask::parse_value(&context, &mut parser).is_err());
+}
+
+#[test]
+fn mask_shorthand_should_parse_origin_and_clip_correctly() {
+    let url = Url::parse("http://localhost").unwrap();
+    let context = ParserContext::new(Origin::Author, &url, Box::new(CSSErrorReporterTest));
+    let mut parser = Parser::new("padding-box content-box");
+    let result = mask::parse_value(&context, &mut parser).unwrap();
+
+    assert_eq!(result.mask_origin.unwrap(), parse_longhand!(mask_origin, "padding-box"));
+    assert_eq!(result.mask_clip.unwrap(), parse_longhand!(mask_clip, "content-box"));
+
+    let mut parser = Parser::new("padding-box padding-box");
+    let result = mask::parse_value(&context, &mut parser).unwrap();
+
+    assert_eq!(result.mask_origin.unwrap(), parse_longhand!(mask_origin, "padding-box"));
+    assert_eq!(result.mask_clip.unwrap(), parse_longhand!(mask_clip, "padding-box"));
+
+    let mut parser = Parser::new("padding-box");
+    let result = mask::parse_value(&context, &mut parser).unwrap();
+
+    assert_eq!(result.mask_origin.unwrap(), parse_longhand!(mask_origin, "padding-box"));
```

Comment:
File a followup about the clip/origin issue and leave a comment here linking to that.